hermes/src/main/java/in/raveesh/hermes/receivers/HermesBroadcastReceiver.java

```diff
@@ -1,38 +1,36 @@
 package in.raveesh.hermes.receivers;
 
-import android.content.BroadcastReceiver;
+import android.app.Activity;
+import android.content.ComponentName;
 import android.content.Context;
 import android.content.Intent;
-import android.util.Log;
+import android.support.v4.content.WakefulBroadcastReceiver;
 
-import in.raveesh.hermes.Hermes;
+import in.raveesh.hermes.Util;
 
-public class HermesBroadcastReceiver extends BroadcastReceiver {
+public class HermesBroadcastReceiver extends WakefulBroadcastReceiver {
 
+    private Class intentService;
     public HermesBroadcastReceiver() {
     }
 
-    protected boolean gotRegistrationId = false;
-    protected boolean error = false;
-    protected String registrationID;
+    public void setIntentService(Class intentService) {
+        this.intentService = intentService;
+        Util.log("Setting intentService class to - "+intentService);
+    }
 
     @Override
     public void onReceive(Context context, Intent intent) {
-        registrationID = intent.getStringExtra("registration_id");
-        if (registrationID != null) {
-            gotRegistrationId = true;
-            if (Hermes.CALLBACK != null) {
-                Hermes.CALLBACK.registrationComplete(registrationID);
-            }
-        } else if (intent.getExtras() != null && intent.getExtras().get("error") != null) {
-            error = true;
-            if (Hermes.CALLBACK != null){
-                Hermes.CALLBACK.registrationFailed(intent.getExtras().get("error").toString());
-            }
-            ExponentialBackoffReceiver.attemptRegistration(context, Hermes.getDelay(), Hermes.getSenderId());
-        } else {
-            Log.d(Hermes.TAG, "GCM Message Received");
-
-        }
+
+        if(intentService == null ) throw new UnsupportedOperationException(
+                "Please use setIntentService in the " +
+                        "constructor of your broadcastReceiver class.");
+
+        ComponentName comp = new ComponentName(
```

Comment:
Fair enough, I haven't read through code in detail, could you please add comments?